src/mtype.c

```diff
@@ -6516,10 +6516,23 @@ void TypeQualified::resolveHelper(Loc loc, Scope *sc,
             // If the symbol is an import, try looking inside the import
             if (Import *si = s->isImport())
             {
-                s = si->search(loc, s->ident);
-                if (s && s != si)
-                    goto L1;
-                s = si;
+                if (idents.dim == 0)
+                {
+                    /* Special case for runnable/Same.d:
```

Comment:
What's extremely weird is (a hidden feature?) that this works with base classes, but not with functions or when directly using the symbol.
```d
module foo;
void foo() {}
```
```
module bar;
import foo;
void test() { foo(); } // Error: function expected before (), not module foo of type void
pragma(msg, foo); // Error: cannot interpret module foo at compile time
```